python/segyio/segy.py

```diff
@@ -302,6 +302,23 @@ def trace(self):
             Fill the file with one trace (filled with zeros)::
                 >>> tr = np.zeros(f.samples)
                 >>> f.trace = itertools.repeat(tr)
+
+            For advanced users: sometimes you want to load the entire segy file
+            to memory and apply your own structural manipulations or operations
+            on it. Some segy files are very large and may not fit, in which
+            case this feature will break down. This is an optimisation feature.
+            If you're doing something trace-by-trace and performing non-trivial
```

Comment:
I think the non-trivial memory intensive bit does not make sense. But maybe if you work trace-by-trace or line-by-line...